iothub/device/src/IotHubModuleClient.cs

```diff
@@ -40,6 +40,8 @@ public class IotHubModuleClient : IotHubBaseClient
         /// <exception cref="InvalidOperationException">Required key/value pairs were missing from the connection string.</exception>
         /// <returns>A disposable client instance.</returns>
         /// <exception cref="InvalidOperationException">A module Id was missing in the connection string. <see cref="IotHubDeviceClient"/> should be used for devices.</exception>
+        /// <exception cref="InvalidOperationException">Different gateway hostnames were specified through the connection string and <see cref="IotHubClientOptions.GatewayHostName"/>.
+        /// It is recommended to not hand edit connection strings but instead use <see cref="IotHubClientOptions"/> to specify values for the additional fields.</exception>
```

Comment:
```suggestion
        /// It is recommended to use <see cref="IotHubClientOptions"/> to specify values for the additional fields rather than hand edit connection strings.</exception>
```